src/MLOps.NET.Azure/Storage/StorageAccountMetaDataStore.cs

```diff
@@ -69,5 +73,50 @@ private async Task<CloudTable> GetTable(string tableName)
 
             return result.Result as TEntity;
         }
+
+        private async Task<IEnumerable<TEntity>> RetrieveEntities<TEntity>(string propertyName, Guid propertyValue, string tableName) where TEntity : ITableEntity, new()
```

Comment:
I think we should a more general query that takes row and partition key to find entities.